server/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -258,7 +260,7 @@ private RoutedCollectPhase createPhase(PlannerContext plannerContext, java.util.
                 ? Lists2.map(boundOutputs, DocReferences::toSourceLookup)
                 : boundOutputs,
             Collections.emptyList(),
-            where.queryOrFallback(),
+            Optimizer.optimizeQuery(plannerContext, where.queryOrFallback()),
```

Comment:
I wonder if we should trigger it inside the `WhereClauseOptimizer` instead to avoid to trigger it in 3 (selete/update/delete) plans.